### ISSUE TITLE
Command model: swap the from-scratch net for a pretrained encoder

### ISSUE THREAD

**Issue by lucaromagnoli** [Conceptual-Machines/magda-core#1847]
## Goal

A **local specialised model that is smart where it needs to be, and not
expensive to run**. Size and latency are constraints to stay within reason, not
things to optimise for. The current command model optimises hard for the wrong
one of those, and it shows.

## What's actually wrong

The on-device command model (#1827) fails on ordinary input:

```
"track with @fm_0"           -> track(name="").rack.new().fx.add(name="<fm_0>")
"new track with @polysynth"  -> track(name="").rack.new().fx.add(name="<polysynth>")
```

An empty name with no `new=true` resolves to the *selected* track, so the user
gets a rack on their current track instead of a new track. Other results from
the same quick sweep:

```
"put @serum on a new track"  -> track(name="").track.set(colour="None")
"i want a track with @massive" -> track(name="I", new=true).fx.add(...)
"track with serum on it"     -> track(name="It", new=true)
```

Root cause of the rack case specifically: `gen_create_with_plugins` only ever
generated *named* tracks ("new bass track with @serum"). There was no unnamed
track-with-plugin example in 24k rows, while `gen_create_rack` does have the
unnamed shape ("new rack with @serum") — so the nearest template wins.

**That is fixable with more templates. The pattern behind it is not.** A
51k-parameter net trained from scratch on synthetic data has no language prior:
it does not know "track" and "rack" are different words, only which template
they co-occurred in. Every distinction has to be hand-authored, and each gap
costs a template edit, a regeneration, a retrain, an export and a new parity
fixture. That is unbounded whack-a-mole against open-ended user input.

## The measurement problem

The model reports **111/111** on the committed test set. That number is
in-distribution: `eval/testset.jsonl` is authored from the same case list the
templates come from. **It has never been measured out-of-distribution.** A
14-phrase sweep written without looking at the templates scored roughly 9/14.

This is the same illusion the router POC hit — 100% on a val split drawn from
its own templates, 59.3% on a hand-authored held-out set. There it was caught
because a separate held-out set existed. Here it wasn't.

## Why the current design looks like this

Not an accident — `findings.md` decision #3 records it:

> A generative model emits DSL token-by-token = an autoregressive loop with KV
> cache + sampling + feedback — exactly what FINN/hls4ml/FPGA dataflow do NOT do.

Every uncomfortable property is downstream of one constraint, *must run on FPGA
fabric*: no autoregression, convs rather than attention, English-only (a
multilingual embedding table does not fit on-chip), ~100-token vocab. Drop the
FPGA constraint from the **CPU product** and none of those justifications hold.

The FPGA target is still valid as its own track — it just should not be setting
the architecture for what ships to users.

## Options

**1. Generative, emits DSL.** The original plan; `model/train_colab.ipynb`
already scaffolds unsloth LoRA on Qwen2.5-0.5B-Instruct exporting GGUF, and
llama.cpp is vendored. Needs GBNF constraining to guarantee valid DSL.
Autoregressive, so the slowest. Worth it only if the model should compose DSL
shapes it was never trained on.

**2. Generative, emits tags.** Keeps the deterministic reconstructor. But it
pays for an autoregressive decode to produce a fixed-length label sequence and
can still emit a malformed tag sequence. Strictly worse than 3 for a closed
label set.

**3. Pretrained encoder + the existing two heads.** Recommended. Same
architecture shape as today, only the encoder is replaced:

```
now:  random embedding + 3 dilated convs  -> mean-pool -> intent head
                                          -> per-token -> BIO slot head
new:  pretrained transformer encoder      -> mean-pool -> intent head
                                          -> per-token -> BIO slot head
```

`dataset/tagging.py`, `reconstruct()`, `magda_dsl/dsl.py`, the grammar and
`InstructionExecutor` are all untouched.

## Why option 3

- **Language prior.** track vs rack stops being something we hand-author.
- **Single forward pass.** Still deterministic, no sampling, no KV cache.
- **Keeps the correctness guarantee.** The renderer builds the DSL from tags, so
  malformed DSL stays structurally impossible. Option 1 gives that up and buys
  it back with GBNF.
- **No new dependency.** ONNX Runtime is already vendored
  (`third_party/onnxruntime`, fetched per platform) and already ships models —
  `resources/models/basic_pitch.onnx` for audio-to-MIDI, plus Spleeter for
  stems. This is an ONNX export into an existing runtime.
- **Multilingual for free** with XLM-RoBERTa, which likely supersedes #1846
  (language packs) rather than needing per-language models.
- Open-vocab names still work — a pretrained encoder is *better* at spotting a
  name span from context than one that learned names from 23 template values.

Cost: XLM-R base is 278M params, ~280 MB int8, ~10-30 ms on CPU. MiniLM-class
encoders are ~5 ms if that turns out to matter. Both are well inside "not too
expensive"; neither is optimising for size.

The one real detail: transformers tokenize to subwords while BIO labels are
per-word. Standard fix — label the first subword of each word and mask the rest.
`tagging.py` already emits word-level tags, so it is a data-loader change.

## Do this first, before any C++ moves

1. **Build an out-of-distribution test set** — 40-60 phrasings written the way
   people actually type, deliberately *not* derived from the templates, held out
   and committed. Score the current model on it. That is the real baseline;
   111/111 is not.
2. **Swap the encoder in the POC only**, train on the existing 24k rows, score
   on the same set. If the language prior does not fix rack/track, that is worth
   knowing before touching the runtime.

Nothing below starts until those two numbers exist.

## Work

- [ ] `eval/ood_testset.jsonl` — hand-authored out-of-distribution set, committed
- [ ] Baseline the shipped model on it (expect a large drop from 111/111)
- [ ] Encoder swap in `model/net.py` + subword/word label alignment in
      `model/data.py`; pick between XLM-R base, mDeBERTa-v3 and a distilled
      multilingual encoder on measured accuracy
- [ ] Compare against the baseline on the OOD set, per intent
- [ ] ONNX export + an ORT-backed `CommandModel` alongside the current one
- [ ] Parity harness locking C++ to the Python reference, as
      `tests/test_command_model.cpp` does now
- [ ] Decide the fate of the conv net: keep as the FPGA track, or retire

## Out of scope

- **The router (#1843) stays as it is.** Seven labels, no span extraction, a
  genuinely closed surface, 88.9% on its held-out set, sub-millisecond. Loading
  a 278M encoder to choose among seven options would be daft. The tiny-net
  approach is correct there; this issue is only about the command model.
- **FPGA.** Neither a transformer encoder nor a generative model maps to FINN.
  If the FPGA target stays live it keeps the conv net, decoupled from CPU.

## Related

- #1827 command model C++ port (the thing being reconsidered)
- #1837 rack + clip-operator intents
- #1843 on-device router (staying tiny)
- #1846 command-model language packs — likely superseded if the encoder is
  multilingual
